Codebase Restructure

kartikcho opened the pull request:
Moved code out of App.js into separate files and folders for better organization and structure.

### TODO

- [x] Run prettier

kartikcho committed:
Move constants to constants.js

kartikcho committed:
Reorg, typo fixes

kartikcho committed:
Move more helper funcs to canvasUtils

kartikcho committed:
Move canvas code to canvas

kartikcho committed:
[WIP] organize helper functions into utils

taneliang commented on src/canvas/renderCanvas.js:
```diff
@@ -0,0 +1,480 @@
+//                   /
+//                  '──────────────────────────
+//
+// TODO Passing "state" directly breaks memoization for e.g. mouse moves
```
This TODO for `renderCanvas` should still be put with `renderCanvas`

kartikcho committed:
Revert "[WIP] organize helper functions into utils"

This reverts commit 99a384962359f8798ade9a98f078177b2e3c29a1.

kartikcho committed:
Move helper functions into utils

kartikcho committed:
Run prettier, fix CLI script command

taneliang commented on src/tracerbench/trace/index.js:
```diff
@@ -8,4 +8,4 @@ export { loadTrace } from './load_trace';
 export { networkConditions, IConditions, INetworkConditions, } from './conditions';
 export * from './trace_event';
-export { getBrowserArgs } from './utils';
+export { getBrowserArgs } from '../../util/utils';
```
`getBrowserArgs` doesn't exist

taneliang commented on src/App.js:
```diff
@@ -42,11 +52,28 @@ import {
   MARKER_TICK_HEIGHT,
+  REACT_PRIORITIES,
+  ROW_CSS_PIXELS_HEIGHT,
+  TEXT_CSS_PIXELS_OFFSET_START,
```
A lot of these constants aren't being used. It may be a good idea to not import these. I think we can leave them in constants.js and clean them up in the future.

taneliang reviewed (commented):
Quite a few import paths are broken; you may want to check those again.

I think we should also rename util/utils.js to util/getBatchRange.js since we now have a util folder.

This PR is too big to review so I haven't looked at the code in detail, but the app still seems to work.

But apart from that (and the comments below) this is a good big first step i the right direction. I think the organization is still not ideal but we can fix that in the future when this feels messy again.

App.js will likely need to be split up further as our profiler will have more than one page, but we can do that in another PR so that we can land this quickly.

Requesting review from @jevakallio to see if he has any tips for us :)

kartikcho committed:
remove unused imports in app.js, rename utils.js

kartikcho commented:
Most of the broken imports were caused by vscode import updates (which is why the earlier commit was failing), apparently `speedscope` had it's own `utils.js` and tracerbench has references to utils too (which doesn't exist as you mentioned in the comment above). 
I've fixed almost all of them and renamed `utils.js` but let me know if you find some left out.

kartikcho committed:
Merge branch 'master' into reorg-and-cleanup

taneliang commented on src/canvas/renderCanvas.js:
```diff
@@ -213,8 +211,9 @@ export const renderReact = ({
 };
 
-import type { PanAndZoomState } from '../usePanAndZoom';
+import type { PanAndZoomState } from '../util/usePanAndZoom';
```
These imports should be at the top of the file. We should add a linter asap to catch these things automatically (I've opened #16)

kartikcho replied:
Agreed, we can update issues like these in a PR that adds linter and fixes said linter errors. For now, let's leave it in.

taneliang replied:
I think we should fix this here, because a random import in the middle of a file is very surprising. We can fix less obvious issues later with the linter's help but since this is new code we should try to keep a certain standard haha.

kartikcho replied:
Alright, done! (Also did an in browser commit, don't @ me)

taneliang reviewed (approved):
> apparently speedscope had it's own utils.js and tracerbench has references to utils too (which doesn't exist as you mentioned in the comment above).

Uhoh. Let's hope we can just black box speedscope and tracerbench (since they're third party code) and hope those missing imports don't break anything.

Approving, but let's wait for @jevakallio's review before merging.

kartikcho committed:
Update renderCanvas.js

kartikcho committed:
prettier CLI fix, move constants to canvas

jevakallio commented on src/canvas/canvasUtils.js:
```diff
@@ -0,0 +1,273 @@
+// @flow
```
I recommend using `git mv` to move a file  and commit it, since now I can't see if the file has been changed or just moved. This way we also lose the version history.

No need to change anything now, only for future reference.

kartikcho replied:
I created this file in this PR but I'll keep that in mind, thanks!

jevakallio reviewed (approved):
LGTM! Did not review renderCanvas and canvasUtils in detail.
